tests/ui/issues/issue-100605.stderr

```diff
@@ -40,6 +34,10 @@ note: function defined here
    |
 LL | fn takes_option(_arg: Option<&String>) {}
    |    ^^^^^^^^^^^^ ---------------------
+help: try using `.as_ref()` to convert `&Option<String>` to `Option<&String>`
+   |
+LL |     takes_option(&res.as_ref());
```

Comment:
This suggestion is not right either. I wonder if there's logic for peeling addr_ofs off of exprs for suggestions :thinking: